.github/workflows/ompi_mpi4py.yaml

```diff
@@ -0,0 +1,109 @@
+name: GitHub Action CI
+
+on: [pull_request]
+
+jobs:
+  mpi4py:
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    steps:
+    - name: Configure hostname
+      run:  echo 127.0.0.1 `hostname` | sudo tee -a /etc/hosts > /dev/null
+      if:   ${{ runner.os == 'Linux' || runner.os == 'macOS' }}
+        
+    - name: Install depencencies
+      run:  sudo apt-get install -y -q
+              libnuma-dev
+      
+    - name: Checkout Open MPI
+      uses: actions/checkout@v4
+      with:
+       path: mpi-build
+       submodules: recursive
+
+    - name: Bootstrap Open MPI
+      run:  ./autogen.pl
+      working-directory: mpi-build
+
+    - name: Configure Open MPI
+      run: ./configure
+              --disable-dependency-tracking
+              --disable-sphinx
+              --disable-man-pages
+              --disable-mpi-fortran
```

Comment:
i'll do that in a subsequent PR.

Reply:
FWIW: I agree with @dalcinl -- it would probably be good to enable Fortran.

Also -- should we try to leverage multiple GitHub actions together?  E.g., I think that adding this Github Action means we're going to have (at least?) 2 builds occurring in Github Actions (i.e., @edgargabriel's RoCM build).  Instead, shouldn't we:

1. Have a GitHub action that does a build that includes everything we want to test in Github Actions (e.g., RoCM, whatever we want/need for mpi4py, ... etc.).
2. Have subsequent pipelined Github actions that use the build from step 1 to run tests on that build.  E.g., have a 2nd Github action to run @edgargabriel's RoCM tests, and have a 3rd Github action to run the mpi4py tests.

That way, we only build once and save Github action minutes.  The overall runtime might also end up being lower, too.

Reply:
I don't necessarily have strong feelings for one solution over another.  I am not convinced that a single build is always feasible, since sometimes packages are conflicting and cannot be installed simultaneously (e.g I would not be surprised if a distro manager doesn't let you install graphics drivers for AMD as well as other graphics card manufacturers, or if there are some dependency conflicts). 

Regarding running jobs, in the past that was not an option on github since they didn't have any instances with AMD GPUs. That might have changed recently (i.e. I know that Azure has instances with AMD GPUs meanwhile), but I haven't investigated whether that would be an option for github workflows.

Reply:
@edgargabriel If you manage to install CuPy within the runner, then you should definitely try to run mpi4py tests with `mpiexec -n <np> python test/main.py --cupy`. Every MPI routine will be tested for both CPU and GPU arrays.